Failure in wallet_balance.py

Failure observed in #14047, output:
```
82/119 - wallet_balance.py failed, Duration: 13 s
stdout:
2019-05-11T08:00:06.629000Z TestFramework (INFO): Initializing test directory /tmp/test_runner_₿_🏃_20190511_074808/wallet_balance_37
2019-05-11T08:00:07.651000Z TestFramework (INFO): Check that only node 0 is watching an address
2019-05-11T08:00:07.658000Z TestFramework (INFO): Mining blocks ...
2019-05-11T08:00:11.525000Z TestFramework (INFO): Test getbalance with different arguments
2019-05-11T08:00:11.708000Z TestFramework (INFO): Test getbalance and getunconfirmedbalance with unconfirmed inputs
2019-05-11T08:00:11.787000Z TestFramework (INFO): Test getbalance and getunconfirmedbalance with conflicted unconfirmed inputs
2019-05-11T08:00:15.377000Z TestFramework (INFO): Check that wallet txs not in the mempool are untrusted
2019-05-11T08:00:15.388000Z TestFramework (INFO): Test replacement and reorg of non-mempool tx
2019-05-11T08:00:18.538000Z TestFramework (INFO): Put txs back into mempool of node 1 (not node 0)
2019-05-11T08:00:18.604000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/home/travis/build/bitcoin/bitcoin/build/bitcoin-x86_64-unknown-linux-gnu/test/functional/test_framework/test_framework.py", line 175, in main
    self.run_test()
  File "/home/travis/build/bitcoin/bitcoin/build/bitcoin-x86_64-unknown-linux-gnu/test/functional/wallet_balance.py", line 203, in run_test
    assert_equal(self.nodes[0].getbalance(minconf=0), 0)  # wallet txs not in the mempool are untrusted
  File "/home/travis/build/bitcoin/bitcoin/build/bitcoin-x86_64-unknown-linux-gnu/test/functional/test_framework/util.py", line 39, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(98.85979500 == 0)
2019-05-11T08:00:18.612000Z TestFramework (INFO): Stopping nodes
```

--- a/test/functional/wallet_balance.py
+++ b/test/functional/wallet_balance.py
@@ -203,6 +203,8 @@ class WalletTest(BitcoinTestFramework):
         self.log.info('Put txs back into mempool of node 1 (not node 0)')
         self.nodes[0].invalidateblock(block_reorg)
         self.nodes[1].invalidateblock(block_reorg)
+        self.sync_blocks()
+        self.nodes[0].syncwithvalidationinterfacequeue()
         assert_equal(self.nodes[0].getbalance(minconf=0), 0)  # wallet txs not in the mempool are untrusted
         self.nodes[0].generatetoaddress(1, ADDRESS_WATCHONLY)
         assert_equal(self.nodes[0].getbalance(minconf=0), 0)  # wallet txs not in the mempool are untrusted

